### PR TITLE
[Enhancement] disallow using persistent index in shared data mode when create table

### PR DESCRIPTION
**Pull request by luohaha**
disallow using persistent index in shared data mode when create table

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|1|2|50.00%|[296]|

**Review by kevincai — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java [hunk 1]
```diff
@@ -291,6 +291,11 @@ public Table createTable(LocalMetastore metastore, Database db, CreateTableStmt
         boolean enablePersistentIndex =
                 PropertyAnalyzer.analyzeBooleanProp(properties, PropertyAnalyzer.PROPERTIES_ENABLE_PERSISTENT_INDEX,
                         false);
+        // analyze cloud native table
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java:294]:
```suggestion
```

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|1|2|50.00%|[295]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**